src/NLog/Layouts/LayoutParser.cs

```diff
@@ -457,8 +461,12 @@ private static void SetDefaultPropertyValue(ConfigurationItemFactory configurati
             }
             else
             {
-                // TODO NLog 5.0 Should throw exception when invalid configuration (Check throwConfigExceptions)
-                InternalLogger.Warn("{0} has no default property to assign value {1}", layoutRenderer.GetType(), value);
+                var configException = new NLogConfigurationException($"{layoutRenderer.GetType()} has no default property to assign value {value}");
+                if (throwConfigExceptions ?? configException.MustBeRethrown())
+                {
+                    throw configException;
+                }
+                InternalLogger.Warn(configException.Message);
```

Comment:
`InternalLogger.Warn(configException.Message);` is not necessary, since `configException.MustBeRethrown()` will perform logging.